torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1460,6 +1476,13 @@ def test_profiler_with_sync_rpc_udf(self):
         self._profiler_test_with_rpc(RPCExecMode.SYNC, my_sleep_func, args=(1,),
                                      use_record_function=True)
 
+    @single_threaded_process_group_agent
+    @dist_init
+    def test_profiler_with_sync_rpc_udf_single_threaded(self):
```

Comment:
In the tests above you renamed the test method to a helper function and then created two new test methods, with different decorators, which invoked that helper function. However in this test case, and in the one that follows, you duplicated the test code. To avoid duplication could we also use the helper method approach in these cases?

Reply:
@lw I see, so do you mean that the following lines:

```
self._profiler_test_with_rpc(RPCExecMode.SYNC, my_sleep_func, args=(1,))
self._profiler_test_with_rpc(RPCExecMode.SYNC, my_sleep_func, args=(1,),
                                     use_record_function=True)
```

should be deduped  and turned into a single test function?

Reply:
Yes, exactly. This test case here is rather short so de-duplicating doesn't bring a huge advantage, but as we go further down in this file the test cases become longer and more complex.

What I am suggesting is, for example, to do this:
```
def _profiler_with_sync_rpc_udf_helper(self):
    self._profiler_test_with_rpc(RPCExecMode.SYNC, my_sleep_func, args=(1,))	                
    self._profiler_test_with_rpc(RPCExecMode.SYNC, my_sleep_func, args=(1,),
                                 use_record_function=True)

@dist_init
def test_profiler_with_sync_rpc_udf(self):
    self._profiler_with_sync_rpc_udf_helper()

@single_threaded_process_group_agent
@dist_init
def test_profiler_with_sync_rpc_udf_single_threaded(self):
    self._profiler_with_sync_rpc_udf_helper()
```

And a similar thing for the tests that follow.

Reply:
Sounds good, just updated.